client/src/logic/game/tasks/action-logic-saga.ts

```diff
@@ -42,18 +42,18 @@ function* actionLogicSaga(gameState: LocalGameState): SagaIterator {
 	const pState = gameState.players[playerId]
 	const lastActionResult = gameState.lastActionResult
 
-	if (gameState.currentModalId !== null) {
-		const id = gameState.currentModalId
+	if (gameState.currentModalData && gameState.currentModalData.modalId) {
+		const id = gameState.currentModalData?.modalId
 		if (id === 'grian_rare') {
 			yield fork(borrowSaga)
```

Comment:
One more thing, actually - can you see how here a huge portion of this if/else is now meaningless? Unless if it's grian, for which we have a custom saga, it should just be setOpenedModal(id)

Reply:
i'll take a look

Reply:
cool fixed